### PR TITLE
fix: server will ignore last data when Read return n,EOF

### PR DESCRIPTION
**Pull request by wzekin**
#### What type of PR is this?

fix
#### What this PR does / why we need it (English/Chinese):
当 `bodyStream.Read` 返回 n, EOF 的时候，server 会直接丢弃掉这次 Read 的数据

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -1834,26 +1834,29 @@ func (sc *serverConn) newWriterAndRequestNoBody(st *stream) (*responseWriter, er
...
 		vbuf := utils.CopyBufPool.Get()
 		buf := vbuf.([]byte)
 		for {
-			n, err := reqCtx.Response.BodyStream().Read(buf)
-			if err == io.EOF {
+			var n int
+			n, err = reqCtx.Response.BodyStream().Read(buf)
+			if n == 0 {
+				if err == nil {
+					panic("BUG: io.Reader returned 0, nil")
```
Comment (Duslia) [inline server.go]: instead of panic, return an error.

Comment (wzekin) [reply server.go]: done

File: server.go [hunk 2]
```diff
@@ -1834,26 +1834,29 @@ func (sc *serverConn) newWriterAndRequestNoBody(st *stream) (*responseWriter, er
...
 		vbuf := utils.CopyBufPool.Get()
 		buf := vbuf.([]byte)
 		for {
-			n, err := reqCtx.Response.BodyStream().Read(buf)
-			if err == io.EOF {
+			var n int
+			n, err = reqCtx.Response.BodyStream().Read(buf)
+			if n == 0 {
+				if err == nil {
+					return errors.New("BUG: io.Reader returned 0, nil")
```
Comment (Duslia) [inline server.go]: errors.New("response bodyStream().Read(buf) returns 0, nil")

Comment (wzekin) [reply server.go]: done~

File: server.go [hunk 3]
```diff
@@ -1834,26 +1834,29 @@ func (sc *serverConn) newWriterAndRequestNoBody(st *stream) (*responseWriter, er
...
 
-func writeResponseBody(rw *responseWriter, reqCtx *app.RequestContext) error {
+func writeResponseBody(rw *responseWriter, reqCtx *app.RequestContext) (err error) {
 	if reqCtx.Response.IsBodyStream() {
 		vbuf := utils.CopyBufPool.Get()
 		buf := vbuf.([]byte)
 		for {
-			n, err := reqCtx.Response.BodyStream().Read(buf)
-			if err == io.EOF {
+			var n int
```
Comment (welkeyever) [inline server.go]: Move this line out of `for` loop?

Comment (wzekin) [reply server.go]: done~

**Review by welkeyever — APPROVED** [review]

**Pull request merged**